### PR TITLE
added public announcements route

### PR DESCRIPTION
**Pull request by varunthakkar1**
## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/XXXXXXXX/pulses/XXXXXXX)

This adds a new route for public announcements needed for announcements page.

## This PR

Exposes GET request for all announcements to public route.

## Screenshots

[Endpoint picture](https://ibb.co/6bMH701)

## Verification Steps

Start the server and verify that you are able to access the new announcements route. I would suggest adding test data with pgAdmin. Also verify that the protected announcement routes are still working as expected.

### CODE REVIEW

File: service/src/test/java/com/codeforcommunity/processor/AnnouncementsProcessorImplTest.java [hunk 1]
```diff
@@ -46,78 +45,6 @@ public void setup() {
             myJooqMock.getContext(), new Emailer(myJooqMock.getContext()));
   }
 
-  // test getting announcements with range covering no events
```
Comment (rymaju) [inline service/src/test/java/com/codeforcommunity/processor/AnnouncementsProcessorImplTest.java:49]: Do we need to delete all our old tests? I'm wondering if we can reuse from for the public route?

Comment (rymaju) [reply service/src/test/java/com/codeforcommunity/processor/AnnouncementsProcessorImplTest.java:49]: nvm you already do that below in a different file

**Review by rymaju — COMMENTED** [review]
I'm worried that by removing the protected route we will be breaking some functionality on the current Vue site. Otherwise everything looks great!

Could I get some confirmation that the route works, like a screenshot of a request working successfully?

File: service/src/main/java/com/codeforcommunity/processor/PublicAnnouncementsProcessorImpl.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  @Override
+  public GetAnnouncementsResponse getAnnouncements(GetAnnouncementsRequest request) {
+    int count = request.getCount();
+    Timestamp start = request.getStartDate();
+    Timestamp end = request.getEndDate();
+
+    List<Announcements> announcements =
+        db.selectFrom(ANNOUNCEMENTS)
+            .where(ANNOUNCEMENTS.CREATED.between(start, end))
+            .and(ANNOUNCEMENTS.EVENT_ID.isNull())
```
Comment (jackblanc) [inline service/src/main/java/com/codeforcommunity/processor/PublicAnnouncementsProcessorImpl.java:35]: Why event_id == null?

**Review by jackblanc — COMMENTED** [review]
New code looks good, definitely don't want to delete the existing route as that's being called by production now.

**Review by rymaju — APPROVED** [review]
LGTM 👍

**Pull request merged**